src/api/Nhs.Appointments.Api/Models/TypeFixDocumentFilter.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Abstractions;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nhs.Appointments.Api.Models;
+
+public class TypeFixDocumentFilter(IEnumerable<Type> typesToFix, TimeProvider time) : IDocumentFilter
+{    
+    public void Apply(IHttpRequestDataObject req, OpenApiDocument document)
+    {
+        foreach (var schema in document.Components.Schemas)
+        {
+            var type = typesToFix.SingleOrDefault(t => t.Name.ToLower() == schema.Key.ToLower());
+            if (type == null)
+                continue;
+
+            if (schema.Value.Type == "object")
+            {
+                var updatedSchema = new OpenApiSchema
+                {
+                    Type = "object",
+                    Properties = new Dictionary<string, OpenApiSchema>()
+                };
+
+                var clrProperties = type.GetProperties();
+
+                foreach (var prop in schema.Value.Properties)
+                {
+                    var targetProperty = clrProperties.Single(p => p.Name.ToLower() == prop.Key.ToLower());
+                    updatedSchema.Properties.Add(prop.Key, FixupSchema(targetProperty.PropertyType, prop.Value));                    
+                }
+
+                document.Components.Schemas[schema.Key] = updatedSchema;
+            }
+        }
+    }
+
+    private OpenApiSchema FixupSchema(Type type, OpenApiSchema originalSchema) 
+    {        
+        var underlyingType = Nullable.GetUnderlyingType(type) ?? type;
+
+        if(underlyingType.IsEnum)
+        {
+            return new OpenApiSchema
+            {
+                Type = "string",
+                Enum = type.GetEnumNames().Select(n => new OpenApiString(n)).Cast<IOpenApiAny>().ToList(),
+                Default = new OpenApiString(type.GetEnumNames().Last())
+            };
+        }
+        else if(underlyingType == typeof(DateOnly))
+        {
+            return new OpenApiSchema
+            {
+                Type = "string",
+                Format = "date",
+                Default = new OpenApiString(time.GetLocalNow().ToString("yyyy-MM-dd"))
+            };
+        }
+        else if(underlyingType == typeof(TimeOnly))
+        {
+            return new OpenApiSchema
+            {
+                Type = "string",
+                Format = "time",
+                Default = new OpenApiString("09:00")
```

Comment:
Perhaps this is going too far, but I could we somehow communicate here that this is a 24 hour time? I thought about switching it to 17:00 or something, but then it's not clear if 9:00 is valid as opposed to 09:00. 

Looks like the `OpenApiSchema` type accepts a string `Description` field, perhaps we could add some text in there?

Reply:
Done